Source/WebKit/WebProcess/Plugins/PDF/PDFPluginScriptEvaluator.mm

```diff
@@ -0,0 +1,220 @@
+        if (!CGPDFDictionaryGetName(javaScriptAction, "S", &actionType) || strcmp(actionType, "JavaScript"))
+            continue;
+
+        const UInt8* bytes = nullptr;
+        CFIndex length = 0;
+        CGPDFStreamRef stream;
+        CGPDFStringRef string;
+        RetainPtr<CFDataRef> data;
+        if (CGPDFDictionaryGetStream(javaScriptAction, "JS", &stream)) {
+            CGPDFDataFormat format;
+            data = adoptCF(CGPDFStreamCopyData(stream, &format));
+            if (!data)
+                continue;
+            bytes = CFDataGetBytePtr(data.get());
+            length = CFDataGetLength(data.get());
+        } else if (CGPDFDictionaryGetString(javaScriptAction, "JS", &string)) {
+            bytes = CGPDFStringGetBytePtr(string);
+            length = CGPDFStringGetLength(string);
+        }
+        if (!bytes || !length)
+            continue;
+
+        CFStringEncoding encoding = (length > 1 && bytes[0] == 0xFE && bytes[1] == 0xFF) ? kCFStringEncodingUnicode : kCFStringEncodingUTF8;
+        RetainPtr<CFStringRef> script = adoptCF(CFStringCreateWithBytes(kCFAllocatorDefault, bytes, length, encoding, true));
+        if (!script)
+            continue;
+
+        scripts.append(script);
+    }
+}
+
+static void jsPDFDocInitialize(JSContextRef ctx, JSObjectRef object)
+{
+    PDFPluginScriptEvaluator* evaluator = static_cast<PDFPluginScriptEvaluator*>(JSObjectGetPrivate(object));
+    evaluator->ref();
+}
+
+static void jsPDFDocFinalize(JSObjectRef object)
+{
+    PDFPluginScriptEvaluator* evaluator = static_cast<PDFPluginScriptEvaluator*>(JSObjectGetPrivate(object));
+    evaluator->deref();
+}
+
+JSClassRef PDFPluginScriptEvaluator::jsPDFDocClass()
+{
+    static const JSStaticFunction jsPDFDocStaticFunctions[] = {
+        { "print", jsPDFDocPrint, kJSPropertyAttributeReadOnly | kJSPropertyAttributeDontDelete },
+        { 0, 0, 0 },
+    };
+
+    static const JSClassDefinition jsPDFDocClassDefinition = {
+        0,
+        kJSClassAttributeNone,
+        "Doc",
+        0,
+        0,
+        jsPDFDocStaticFunctions,
+        jsPDFDocInitialize, jsPDFDocFinalize, 0, 0, 0, 0, 0, 0, 0, 0, 0
+    };
+
+    static JSClassRef jsPDFDocClass = JSClassCreate(&jsPDFDocClassDefinition);
+    return jsPDFDocClass;
+}
+
+JSValueRef PDFPluginScriptEvaluator::jsPDFDocPrint(JSContextRef ctx, JSObjectRef function, JSObjectRef thisObject, size_t argumentCount, const JSValueRef arguments[], JSValueRef* exception)
+{
+    if (!JSValueIsObjectOfClass(ctx, thisObject, jsPDFDocClass()))
+        return JSValueMakeUndefined(ctx);
+
+    RefPtr evaluator = static_cast<PDFPluginScriptEvaluator*>(JSObjectGetPrivate(thisObject));
+    evaluator->print();
+
+    return JSValueMakeUndefined(ctx);
+}
+
+void PDFPluginScriptEvaluator::runScripts(CGPDFDocumentRef document, Client& client)
+{
+    ASSERT(isMainRunLoop());
+
+    Ref<PDFPluginScriptEvaluator> evaluator = PDFPluginScriptEvaluator::create(document, client);
```

Comment:
`auto`

Reply:
Actually shouldn't it be `Ref evaluator` now?